### PR TITLE
adding tests for priors

### PR DESCRIPTION
**Pull request by KrisThielemans**
currently failing...

### CODE REVIEW

**Comment by KrisThielemans** [comment]
I currently see the following failures:
- Quadratic prior has edge effects, but aside from that a nearly constant diff between the analytical and numerical gradient
- PLS prior is ok in the middie, but has strong edge effects.

@robbietuk could you try the QuadraticPrior with your Python test?

**Comment by robbietuk** [comment]
My python QP test seems reasonable. I do not have edge effects.

**Comment by KrisThielemans** [comment]
I've brought this up-to-date, but not changed the tests themselves.

**Comment by robbietuk** [comment]
Is it worth rebasing instead? The commit history and file changes are a bit of a mess :/

**Comment by KrisThielemans** [comment]
hmmm. weird. I'll do that.

**Comment by KrisThielemans** [comment]
ok, it was on release_4, with some stuff happening in between. cleaner now

File: src/recon_test/test_priors.cxx [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+  shared_ptr<target_type> gradient_2_sptr(target.get_empty_copy());
+  info("Computing gradient",3);
+  objective_function.compute_gradient(*gradient_sptr, target);
+  this->set_tolerance(std::max(fabs(double(gradient_sptr->find_min())), double(gradient_sptr->find_max()))/1000);
+  info("Computing objective function at target",3);
+  const double value_at_target = objective_function.compute_value(target);
+  target_type::full_iterator target_iter=target.begin_all();
+  target_type::full_iterator gradient_iter=gradient_sptr->begin_all();
+  target_type::full_iterator gradient_2_iter=gradient_2_sptr->begin_all(); 
+  const float eps = 1e-2F;
```
Comment (robbietuk) [inline src/recon_test/test_priors.cxx:117]: This is too large a perturbation

**Comment by robbietuk** [comment]
Would resolve #153

**Comment by robbietuk** [comment]
Hi, I have made some changes to this PR, like including the relative difference prior (which fails) but I do not have permission to push to the branch.

**Comment by KrisThielemans** [comment]
I notice that Travis is green now. thanks for the advice on `eps`.

either do a PR on this one, or just start a new one (might be easiest). depending on what your changes are, we could merge this one first. As you prefer.

could you check the Codacy warning? https://app.codacy.com/manual/KrisThielemans/STIR/pullRequest?prid=5421028

**Comment by robbietuk** [comment]
I'll try merging to your branch and check Codacy. Also while travis is passing, I am not sure it is actually running `test_priors` as I get an error with PLS on my machine.

**Comment by KrisThielemans** [comment]
superseded by #695

**Pull request closed without merging**